src/linear_algebra/matrix/reindex.lean

```diff
@@ -52,14 +57,52 @@ rfl
   reindex_linear_equiv (equiv.refl m) (equiv.refl n) = linear_equiv.refl R _ :=
 linear_equiv.ext $ λ _, rfl
 
-lemma reindex_linear_equiv_mul {o o' : Type*} [fintype o] [fintype o'] [semiring R]
+lemma reindex_linear_equiv_trans [semiring R] (e₁ : m ≃ m') (e₂ : n ≃ n') (e₁' : m' ≃ m'')
+  (e₂' : n' ≃ n'') : (reindex_linear_equiv e₁ e₂).trans (reindex_linear_equiv e₁' e₂') =
+   @reindex_linear_equiv _ _ _ _ _ _ _ _ R _ (e₁.trans e₁') (e₂.trans e₂') :=
+by { ext, refl }
+
+lemma reindex_linear_equiv_comp [semiring R] (e₁ : m ≃ m') (e₂ : n ≃ n') (e₁' : m' ≃ m'')
+  (e₂' : n' ≃ n'') : (@reindex_linear_equiv _ _ _ _ _ _ _ _ R _ e₁' e₂') ∘
+   (@reindex_linear_equiv _ _ _ _ _ _ _ _ R _ e₁ e₂)
+  = (reindex_linear_equiv (e₁.trans e₁') (e₂.trans e₂')) :=
+by { rw [← reindex_linear_equiv_trans], refl }
+
+lemma reindex_linear_equiv_comp_apply [semiring R] (e₁ : m ≃ m') (e₂ : n ≃ n') (e₁' : m' ≃ m'')
+  (e₂' : n' ≃ n'') (M : matrix m n R) :
+  (reindex_linear_equiv e₁' e₂') (reindex_linear_equiv e₁ e₂ M) =
+    reindex_linear_equiv (e₁.trans e₁') (e₂.trans e₂') M :=
+by rw [← reindex_linear_equiv_comp e₁ e₂ e₁' e₂']
+
+lemma reindex_linear_equiv_one [semiring R] [decidable_eq m] [decidable_eq m']
+  (e : m ≃ m') : (reindex_linear_equiv e e (1 : matrix m m R)) = 1 :=
+minor_one_equiv e.symm
+
+variables {o o' : Type*} [fintype o] [fintype o']
+
+lemma reindex_linear_equiv_mul [semiring R]
   (eₘ : m ≃ m') (eₙ : n ≃ n') (eₒ : o ≃ o') (M : matrix m n R) (N : matrix n o R) :
   reindex_linear_equiv eₘ eₒ (M ⬝ N) =
     reindex_linear_equiv eₘ eₙ M ⬝ reindex_linear_equiv eₙ eₒ N :=
 minor_mul_equiv M N _ _ _
 
-/-- For square matrices, the natural map that reindexes a matrix's rows and columns with equivalent
-types, `matrix.reindex`, is an equivalence of algebras. -/
+lemma mul_reindex_linear_equiv_one [semiring R] [decidable_eq o] (e₁ : o ≃ n) (e₂ : o ≃ n')
+  (M : matrix m n R) : M.mul (reindex_linear_equiv e₁ e₂ 1) =
+    reindex_linear_equiv (equiv.refl m) (e₁.symm.trans e₂) M :=
+begin
+  have : M = reindex_linear_equiv (equiv.refl m) e₁ (reindex_linear_equiv (equiv.refl m) e₁.symm M),
+  { rw [reindex_linear_equiv_comp_apply, equiv.symm_trans, equiv.refl_trans,
+      reindex_linear_equiv_refl_refl],
+    refl },
+  conv_lhs { rw this },
+  rw [← reindex_linear_equiv_mul, matrix.mul_one, reindex_linear_equiv_comp_apply,
+    equiv.refl_trans]
+end
```

Comment:
A much shorter proof:
```suggestion
mul_minor_one _ _ _
```

Reply:
Oh, sure, but can you give a hint on how you found it? Otherwise I always end up committing your very nice suggestions, but never improving my Lean-writing... ;-(

Reply:
I found it because I know that `reindex_linear_equiv e1 e2 M` is defeq to `M.minor e1.symm e2.symm`. I think you could have found that without prior knowledge with `dsimp, library_search`

Reply:
Ah ok, thanks. So I gather that you often try and search with `library_search`, right? I somehow find myself very seldom using it. But does it also works in tactic mode or do you need to create a special `example` with your goal as main statement?

Reply:
You can always use `library_search` mid-proof if your goal is in a "surely we have this already" form. In this specific case though, the search was actually just me noticing that this lemma was obviously defeq to the other one in this PR